posts/how-to-commit-code-with-the-git-command-line-interface.md

```diff
@@ -0,0 +1,210 @@
+
+```
+git version 2.32.0
+```
+
+Note that when talking about Git, the word “commit” can be used both as a noun (as in: “Here is the latest commit.”) and a verb meaning “to create a commit” (as in: “I will commit this change.”).
+
+## Step 1 — Checking the repository status
+
+First, it’s useful to get a summary of the changes that we’ve made in our project’s working directory. This will help us select the changes that we want to add to our commit. Run the `git status` command:
+
+```sh
+git status
+```
+
+The output of the command tells us about the changes that are available to be added to our commit. The first section of the output that we’ll pay attention to is titled `Changes not staged for commit`.
+
+```
+Changes not staged for commit:
+  (use "git add <file>..." to update what will be committed)
+  (use "git restore <file>..." to discard changes in working directory)
+        modified:   README.md
+```
+
+This section lists files that already exist in the repository and have been modified since the last commit. It tells us that we have one modified file, `README.md`.
+
+The second section that is relevant to us is titled `Untracked files`. It lists newly created files that don’t yet exist in the repository.
+
+```
+Untracked files:
+  (use "git add <file>..." to include in what will be committed)
+        getting-started.txt
+```
+
+The above output tells us that there’s one new file in the working directory, `getting-started.txt`.
+
+After reading the output of `git status`, we can proceed to select the changes that we want to commit, and we’ll do that by adding those changes to the staging area.
+
+## Step 2 — Selecting changes to be committed
+
+In a Git repository, the **staging area** is the collection of changes that will be included in the next commit. Before we can commit our changes, we have to stage the changes we intend to include in the commit.
+
+### Adding a modified file to the staging area
+
+To add a modified file to the staging area, we use the `git add` command. The `git add` command, in its simplest form, expects a file name as an argument. Here’s an example in which we add the file named `README.md` to the staging area:
+
+```sh
+git add README.md
+```
+
+If we run `git status` again at this point, `README.md` will be gone from the “Changes not staged for commit” section and will instead be present under “Changes to be committed.” This confirms that it has been added to the staging area.
+
+We can continue by adding more files using `git add`. For example, we’ll add the newly created file, `getting-started.txt`:
+
+```sh
+git add getting-started.txt
+```
+
+After this command, `getting-started.txt` will also be included in the “Changes to be committed” section of the `git status` output. By using `git add` in this way, we can include as many changed files as we want in our commit.
+
+### Adding all current repository changes to the staging area
```

Comment:
In this section I would clarify that the `-A` or `--all` flag recursively adds all the modified files in all the directories in the current repo, regardless of where you are running the command from

Reply:
I added this clarification to the last paragraph of the section. I added a mention of `--all`.

Reply:
Thanks, I did a light edit of the last two sentences for clarity and grammar.